src/assignments/phat/lesson_3_answers.clj

```diff
@@ -0,0 +1,143 @@
+(ns lesson-3-answers.core)
+(require '[clojure.string :as str])
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;; 
+;; 128 - Recognize Playing Cards
+
+;; define maps for suit and rank
+;; split input string into 2 parts
+;; first part is a key in suit
+;; second part is a key in rank
+
+(defn recognize-card
+  "Take a string and convert into a map of suit and rank"
+  [card]
+  (let [suit {"D" :diamond 
+              "H" :heart 
+              "C" :club 
+              "S" :spade}
+        card-rank {"2" 0 "3" 1 "4" 2 "5" 3 "6" 4 
+                   "7" 5 "8" 6 "9" 7 "T" 8 "J" 9
```

Comment:
Very nice.  But perhaps you could have simply destructured the 2 chars from the input instead of resorting to string/split?
Also, you're splitting the same string twice; perhaps capture the results in a binding and reuse them?

Finally, (hash-map ....) is typically used with programmatic arguments; here you know exactly what has to be returned, so perhaps just
```
{:suit (suit...)
 :rank (card-rank...)}
```
would suffice?

Reply:
yes. I've fixed it. Thank you!